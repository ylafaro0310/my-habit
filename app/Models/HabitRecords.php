<?php

namespace App\Models;

use Illuminate\Support\Facades\DB;
use Illuminate\Support\Facades\Log;
use App\Models\BaseModel;

class HabitRecords extends BaseModel {
    public $validationRules = [
        'habit_id' => 'required|integer',
        'completed_at' => 'required|integer',
        'is_skipped' => 'required|boolean',
    ];

    public $tableName = 'habit_records';
}