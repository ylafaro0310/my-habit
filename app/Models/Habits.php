<?php

namespace App\Models;

use Illuminate\Support\Facades\DB;
use Illuminate\Support\Facades\Log;

class Habits {
    static $validationRules = [
        'habit_name' => 'required|string|max255',
        'repeat_type' => 'required|string|in:day,week,day_of_week',
        'repeat_value' => 'required|integer',
        'started_at' => 'required|integer',
        'target_time' => 'nullable|integer',
        'time_of_day' => 'required|string|in:always,am,pm',
        'consecutive_days' => 'required|integer',
        'consecutive_weeks' => 'nullable|integer',
    ];

    static $tableName = 'habits';

    function __construct(){
        $this->pdo = DB::connection()->getPdo();
    }
    
    public function index(){
        $sth = $this->pdo->prepare("select * from ".self::$tableName);
        $sth->execute();
        $result = $sth->fetchAll();
        return $result;
    }

    public function store($params){
        $columns = implode(', ',array_keys($params));
        $values = implode(', ',array_map(function($v){return $v===NULL?"NULL":"'$v'";},array_values($params)));
        $sth = $this->pdo->prepare("insert into ".self::$tableName." (${columns}) values (${values})");
        return $sth->execute();
    }

    public function update($id,$params){
        $setValues = array_map(function($key,$value){
            return $value === NULL ? "$key = NULL" : "$key = '$value'";
        },array_keys($params),array_values($params));
        $setValues = implode(', ',$setValues);
        $sth = $this->pdo->prepare("update ".self::$tableName." set $setValues where id = $id");
        return $sth->execute();
    }

    public function destroy($id){
        $sth = $this->pdo->prepare("delete from ".self::$tableName." where id = ${id}");
        return $sth->execute();
    }
}