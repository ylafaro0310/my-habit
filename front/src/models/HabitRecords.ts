import { List, Record } from 'immutable';
import { Dayjs } from 'dayjs';

import dayjs from '../lib/dayjs-ja';
import { JSObject } from '../types/Common';

export class HabitRecord extends Record<{
  habitId: number;
  completedAt: Dayjs;
  isSkipped: boolean;
}>({
  habitId: 0,
  completedAt: dayjs(),
  isSkipped: false,
}) {
  static fromResponse(response: JSObject): HabitRecord {
    const params = { ...response };
    params['completedAt'] = dayjs(response['completedAt'], 'YYYY-MM-DD HH:mm:ss');
    return new HabitRecord(params);
  }
}

export default class HabitRecords extends Record<{
  items: List<HabitRecord>;
}>({
  items: List(),
}) {
  static fromResponse(response: JSObject): HabitRecords {
    const params = { ...response };
    params.items = List(params.habitRecords.map((item: JSObject) => HabitRecord.fromResponse(item)));
    return new HabitRecords(params);
  }
  getList(): List<HabitRecord> {
    return this.get('items');
  }
  filterById(id: number): HabitRecords {
    return this.set(
      'items',
      this.get('items').filter(habitRecord => habitRecord.habitId === id),
    );
  }
  filterByCompletedAt(date: Dayjs): HabitRecords {
    return this.set(
      'items',
      this.get('items').filter(habitRecord => habitRecord.completedAt.isSame(date, 'd')),
    );
  }
}
