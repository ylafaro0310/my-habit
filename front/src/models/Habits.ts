import { List, Record } from 'immutable';
import { Dayjs } from 'dayjs';

import { JSObject } from '../types/Common';
import dayjs from '../lib/dayjs-ja';

import HabitRecords from './HabitRecords';

export class Habit extends Record<{
  id: number;
  habitName: string;
  repeatType: string;
  repeatValue: number;
  startedAt: Dayjs;
  targetTime: number; // 単位は「分」
  timeOfDay: string;
  consecutiveDays: number;
  //consecutiveWeeks: number | null;
}>({
  id: 0,
  habitName: '',
  repeatType: 'day_of_week',
  repeatValue: 127,
  startedAt: dayjs(),
  targetTime: 0,
  timeOfDay: '',
  consecutiveDays: 0,
  //consecutiveWeeks: null,
}) {
  static fromResponse(response: JSObject): Habit {
    const params = { ...response };
    return new Habit(params);
  }
  isCompleted(date: Dayjs, habitRecords: HabitRecords): boolean {
    return habitRecords
      ? habitRecords
          .filterById(this.id)
          .filterByCompletedAt(date)
          .getList().size > 0
      : false;
  }
}

export default class Habits extends Record<{
  items: List<Habit>;
}>({
  items: List(),
}) {
  static fromResponse(response: JSObject): Habits {
    const params = { ...response };
    params.items = List(params.habits.map((item: JSObject) => Habit.fromResponse(item)));
    return new Habits(params);
  }
  getList(): List<Habit> {
    return this.get('items');
  }
}
