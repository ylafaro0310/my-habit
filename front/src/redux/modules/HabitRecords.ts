import actionCreatorFactory from 'typescript-fsa';
import { reducerWithInitialState } from 'typescript-fsa-reducers';

import HabitRecords from '../../models/HabitRecords';

// Action Creator
const actionCreator = actionCreatorFactory('HabitRecord');
export const HabitRecordsActions = {
  getHabitRecords: actionCreator<object>('getHabitRecords'),
  setHabitRecords: actionCreator<HabitRecords>('setHabitRecords'),
  addHabitRecord: actionCreator<object>('addHabitRecord'),
  removeHabitRecord: actionCreator<object>('removeHabitRecord'),
};

// Reducers
export const habitRecordsReducer = reducerWithInitialState(new HabitRecords()).case(
  HabitRecordsActions.setHabitRecords,
  (state, payload) => {
    return state.set('items', payload.getList());
  },
);
