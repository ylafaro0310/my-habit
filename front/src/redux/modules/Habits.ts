import actionCreatorFactory from 'typescript-fsa';
import { reducerWithInitialState } from 'typescript-fsa-reducers';

import Habits from '../../models/Habits';

// Action Creator
const actionCreator = actionCreatorFactory('Habit');
export const HabitsActions = {
  getHabits: actionCreator<void>('getHabits'),
  setHabits: actionCreator<Habits>('setHabits'),
  addHabit: actionCreator<object>('addHabit'),
  updateHabit: actionCreator<{ habitId: number; values: object }>(
    'updateHabit',
  ),
  removeHabit: actionCreator<number>('removeHabit'),
  formInitialize: actionCreator<number>('habitFormInitialize'),
};

// Reducers
export const habitsReducer = reducerWithInitialState(new Habits()).case(
  HabitsActions.setHabits,
  (state, payload) => {
    return state.set('items', payload.getList());
  },
);
