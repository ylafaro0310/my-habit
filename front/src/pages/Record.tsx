import React from 'react';
import styled from 'styled-components';
import { connect } from 'react-redux';
import { Dispatch } from 'redux';

import dayjs from '../lib/dayjs-ja';
import HabitList from '../components/HabitList';
import { State } from '../redux/store';
import { HabitRecordsActions } from '../redux/modules/HabitRecords';
import Habits from '../models/Habits';
import HabitRecords from '../models/HabitRecords';

interface RecordProps {
  habits: Habits;
  habitRecords: HabitRecords;
  dispatch: Dispatch;
}
interface RecordState {
  selectedDate: string;
}
export class Record extends React.Component<RecordProps, RecordState> {
  constructor(props: RecordProps) {
    super(props);
    this.state = {
      selectedDate: dayjs().format('YYYY-MM-DD'),
    };
    this.onChangeDate = this.onChangeDate.bind(this);
  }

  onChangeDate(event: React.ChangeEvent<HTMLInputElement>): void {
    const { value } = event.target;
    this.setState({ selectedDate: value });
  }

  componentDidMount(): void {
    const { dispatch } = this.props;
    dispatch(HabitRecordsActions.getHabitRecords({}));
  }

  render() {
    const { selectedDate } = this.state;
    const { habits, habitRecords } = this.props;
    const date = [];
    const startOfMonth = dayjs().startOf('month');
    const endOfMonth = dayjs().endOf('month');
    for (let i = startOfMonth; !i.isAfter(endOfMonth); i = i.add(1, 'day')) {
      date.push(
        <div>
          <input
            type='radio'
            name='date'
            value={i.format('YYYY-MM-DD')}
            id={i.format('YYYY-MM-DD')}
            checked={dayjs(selectedDate).isSame(i, 'd')}
            onChange={this.onChangeDate}
          />
          <label htmlFor={i.format('YYYY-MM-DD')}>{i.format('MM/DD')}</label>
        </div>,
      );
    }
    return (
      <div className='App'>
        <div className='date'>
          <button>習慣を追加する</button>
          <DateList>
            {date.map((elem, key) => (
              <li key={key}>{elem}</li>
            ))}
          </DateList>
        </div>
        <HabitList habits={habits} habitRecords={habitRecords} selectedDate={selectedDate} />
      </div>
    );
  }
}

export default connect((state: State) => ({
  habits: state.habits,
  habitRecords: state.habitRecords,
}))(Record);

const DateList = styled.ul`
  & li {
    display: inline-block;
  }
`;
