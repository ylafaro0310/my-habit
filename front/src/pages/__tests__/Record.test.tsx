import React from 'react';
import { shallow } from 'enzyme';
import MockDate from 'mockdate';
import dayjs from 'dayjs';

import { Record } from '../Record';
import Habits from '../../models/Habits';
import HabitRecords from '../../models/HabitRecords';

MockDate.set('2020-02-08');

const data = {
  habits: [
    { id: 1, habitName: '読書する', startedAt: dayjs() },
    { id: 2, habitName: '筋トレ', startedAt: dayjs() },
    { id: 3, habitName: '新しいCDを1枚聴く', startedAt: dayjs() },
  ],
  habitRecords: [
    { habitId: 2, completedAt: dayjs() },
    { habitId: 1, completedAt: dayjs('2020-01-08') },
  ],
};

const props = {
  habits: Habits.fromResponse(data),
  habitRecords: HabitRecords.fromResponse(data),
  dispatch: jest.fn(),
};

describe('Page: Record', () => {
  it('Snapshot test', () => {
    const wrapper = shallow(<Record {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
  it('日付をクリックしたらStateが変更されること', () => {
    const wrapper = shallow(<Record {...props} />);
    const yesterday = dayjs()
      .subtract(1, 'day')
      .format('YYYY-MM-DD');
    wrapper.find(`input[value='${yesterday}']`).simulate('change', {
      target: {
        value: yesterday,
      },
    });
    expect(wrapper.state('selectedDate')).toBe(yesterday);
  });
});
