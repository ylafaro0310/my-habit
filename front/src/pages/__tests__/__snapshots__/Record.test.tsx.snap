// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Page: Record Snapshot test 1`] = `
<div
  className="App"
>
  <StyledCard>
    <div
      className="date"
    >
      <styled.div>
        <div>
          2月8日
        </div>
        <styled.a />
      </styled.div>
      <styled.ul
        id="datelist"
      >
        <li
          key="0"
        >
          <styled.div>
            <input
              checked={false}
              id="2020-01-09"
              name="date"
              onChange={[Function]}
              type="radio"
              value="2020-01-09"
            />
            <label
              htmlFor="2020-01-09"
            >
              <div>
                <div>
                  木
                </div>
                <div>
                  9
                </div>
              </div>
            </label>
          </styled.div>
        </li>
        <li
          key="1"
        >
          <styled.div>
            <input
              checked={false}
              id="2020-01-10"
              name="date"
              onChange={[Function]}
              type="radio"
              value="2020-01-10"
            />
            <label
              htmlFor="2020-01-10"
            >
              <div>
                <div>
                  金
                </div>
                <div>
                  10
                </div>
              </div>
            </label>
          </styled.div>
        </li>
        <li
          key="2"
        >
          <styled.div>
            <input
              checked={false}
              id="2020-01-11"
              name="date"
              onChange={[Function]}
              type="radio"
              value="2020-01-11"
            />
            <label
              htmlFor="2020-01-11"
            >
              <div>
                <div>
                  土
                </div>
                <div>
                  11
                </div>
              </div>
            </label>
          </styled.div>
        </li>
        <li
          key="3"
        >
          <styled.div>
            <input
              checked={false}
              id="2020-01-12"
              name="date"
              onChange={[Function]}
              type="radio"
              value="2020-01-12"
            />
            <label
              htmlFor="2020-01-12"
            >
              <div>
                <div>
                  日
                </div>
                <div>
                  12
                </div>
              </div>
            </label>
          </styled.div>
        </li>
        <li
          key="4"
        >
          <styled.div>
            <input
              checked={false}
              id="2020-01-13"
              name="date"
              onChange={[Function]}
              type="radio"
              value="2020-01-13"
            />
            <label
              htmlFor="2020-01-13"
            >
              <div>
                <div>
                  月
                </div>
                <div>
                  13
                </div>
              </div>
            </label>
          </styled.div>
        </li>
        <li
          key="5"
        >
          <styled.div>
            <input
              checked={false}
              id="2020-01-14"
              name="date"
              onChange={[Function]}
              type="radio"
              value="2020-01-14"
            />
            <label
              htmlFor="2020-01-14"
            >
              <div>
                <div>
                  火
                </div>
                <div>
                  14
                </div>
              </div>
            </label>
          </styled.div>
        </li>
        <li
          key="6"
        >
          <styled.div>
            <input
              checked={false}
              id="2020-01-15"
              name="date"
              onChange={[Function]}
              type="radio"
              value="2020-01-15"
            />
            <label
              htmlFor="2020-01-15"
            >
              <div>
                <div>
                  水
                </div>
                <div>
                  15
                </div>
              </div>
            </label>
          </styled.div>
        </li>
        <li
          key="7"
        >
          <styled.div>
            <input
              checked={false}
              id="2020-01-16"
              name="date"
              onChange={[Function]}
              type="radio"
              value="2020-01-16"
            />
            <label
              htmlFor="2020-01-16"
            >
              <div>
                <div>
                  木
                </div>
                <div>
                  16
                </div>
              </div>
            </label>
          </styled.div>
        </li>
        <li
          key="8"
        >
          <styled.div>
            <input
              checked={false}
              id="2020-01-17"
              name="date"
              onChange={[Function]}
              type="radio"
              value="2020-01-17"
            />
            <label
              htmlFor="2020-01-17"
            >
              <div>
                <div>
                  金
                </div>
                <div>
                  17
                </div>
              </div>
            </label>
          </styled.div>
        </li>
        <li
          key="9"
        >
          <styled.div>
            <input
              checked={false}
              id="2020-01-18"
              name="date"
              onChange={[Function]}
              type="radio"
              value="2020-01-18"
            />
            <label
              htmlFor="2020-01-18"
            >
              <div>
                <div>
                  土
                </div>
                <div>
                  18
                </div>
              </div>
            </label>
          </styled.div>
        </li>
        <li
          key="10"
        >
          <styled.div>
            <input
              checked={false}
              id="2020-01-19"
              name="date"
              onChange={[Function]}
              type="radio"
              value="2020-01-19"
            />
            <label
              htmlFor="2020-01-19"
            >
              <div>
                <div>
                  日
                </div>
                <div>
                  19
                </div>
              </div>
            </label>
          </styled.div>
        </li>
        <li
          key="11"
        >
          <styled.div>
            <input
              checked={false}
              id="2020-01-20"
              name="date"
              onChange={[Function]}
              type="radio"
              value="2020-01-20"
            />
            <label
              htmlFor="2020-01-20"
            >
              <div>
                <div>
                  月
                </div>
                <div>
                  20
                </div>
              </div>
            </label>
          </styled.div>
        </li>
        <li
          key="12"
        >
          <styled.div>
            <input
              checked={false}
              id="2020-01-21"
              name="date"
              onChange={[Function]}
              type="radio"
              value="2020-01-21"
            />
            <label
              htmlFor="2020-01-21"
            >
              <div>
                <div>
                  火
                </div>
                <div>
                  21
                </div>
              </div>
            </label>
          </styled.div>
        </li>
        <li
          key="13"
        >
          <styled.div>
            <input
              checked={false}
              id="2020-01-22"
              name="date"
              onChange={[Function]}
              type="radio"
              value="2020-01-22"
            />
            <label
              htmlFor="2020-01-22"
            >
              <div>
                <div>
                  水
                </div>
                <div>
                  22
                </div>
              </div>
            </label>
          </styled.div>
        </li>
        <li
          key="14"
        >
          <styled.div>
            <input
              checked={false}
              id="2020-01-23"
              name="date"
              onChange={[Function]}
              type="radio"
              value="2020-01-23"
            />
            <label
              htmlFor="2020-01-23"
            >
              <div>
                <div>
                  木
                </div>
                <div>
                  23
                </div>
              </div>
            </label>
          </styled.div>
        </li>
        <li
          key="15"
        >
          <styled.div>
            <input
              checked={false}
              id="2020-01-24"
              name="date"
              onChange={[Function]}
              type="radio"
              value="2020-01-24"
            />
            <label
              htmlFor="2020-01-24"
            >
              <div>
                <div>
                  金
                </div>
                <div>
                  24
                </div>
              </div>
            </label>
          </styled.div>
        </li>
        <li
          key="16"
        >
          <styled.div>
            <input
              checked={false}
              id="2020-01-25"
              name="date"
              onChange={[Function]}
              type="radio"
              value="2020-01-25"
            />
            <label
              htmlFor="2020-01-25"
            >
              <div>
                <div>
                  土
                </div>
                <div>
                  25
                </div>
              </div>
            </label>
          </styled.div>
        </li>
        <li
          key="17"
        >
          <styled.div>
            <input
              checked={false}
              id="2020-01-26"
              name="date"
              onChange={[Function]}
              type="radio"
              value="2020-01-26"
            />
            <label
              htmlFor="2020-01-26"
            >
              <div>
                <div>
                  日
                </div>
                <div>
                  26
                </div>
              </div>
            </label>
          </styled.div>
        </li>
        <li
          key="18"
        >
          <styled.div>
            <input
              checked={false}
              id="2020-01-27"
              name="date"
              onChange={[Function]}
              type="radio"
              value="2020-01-27"
            />
            <label
              htmlFor="2020-01-27"
            >
              <div>
                <div>
                  月
                </div>
                <div>
                  27
                </div>
              </div>
            </label>
          </styled.div>
        </li>
        <li
          key="19"
        >
          <styled.div>
            <input
              checked={false}
              id="2020-01-28"
              name="date"
              onChange={[Function]}
              type="radio"
              value="2020-01-28"
            />
            <label
              htmlFor="2020-01-28"
            >
              <div>
                <div>
                  火
                </div>
                <div>
                  28
                </div>
              </div>
            </label>
          </styled.div>
        </li>
        <li
          key="20"
        >
          <styled.div>
            <input
              checked={false}
              id="2020-01-29"
              name="date"
              onChange={[Function]}
              type="radio"
              value="2020-01-29"
            />
            <label
              htmlFor="2020-01-29"
            >
              <div>
                <div>
                  水
                </div>
                <div>
                  29
                </div>
              </div>
            </label>
          </styled.div>
        </li>
        <li
          key="21"
        >
          <styled.div>
            <input
              checked={false}
              id="2020-01-30"
              name="date"
              onChange={[Function]}
              type="radio"
              value="2020-01-30"
            />
            <label
              htmlFor="2020-01-30"
            >
              <div>
                <div>
                  木
                </div>
                <div>
                  30
                </div>
              </div>
            </label>
          </styled.div>
        </li>
        <li
          key="22"
        >
          <styled.div>
            <input
              checked={false}
              id="2020-01-31"
              name="date"
              onChange={[Function]}
              type="radio"
              value="2020-01-31"
            />
            <label
              htmlFor="2020-01-31"
            >
              <div>
                <div>
                  金
                </div>
                <div>
                  31
                </div>
              </div>
            </label>
          </styled.div>
        </li>
        <li
          key="23"
        >
          <styled.div>
            <input
              checked={false}
              id="2020-02-01"
              name="date"
              onChange={[Function]}
              type="radio"
              value="2020-02-01"
            />
            <label
              htmlFor="2020-02-01"
            >
              <div>
                <div>
                  土
                </div>
                <div>
                  1
                </div>
              </div>
            </label>
          </styled.div>
        </li>
        <li
          key="24"
        >
          <styled.div>
            <input
              checked={false}
              id="2020-02-02"
              name="date"
              onChange={[Function]}
              type="radio"
              value="2020-02-02"
            />
            <label
              htmlFor="2020-02-02"
            >
              <div>
                <div>
                  日
                </div>
                <div>
                  2
                </div>
              </div>
            </label>
          </styled.div>
        </li>
        <li
          key="25"
        >
          <styled.div>
            <input
              checked={false}
              id="2020-02-03"
              name="date"
              onChange={[Function]}
              type="radio"
              value="2020-02-03"
            />
            <label
              htmlFor="2020-02-03"
            >
              <div>
                <div>
                  月
                </div>
                <div>
                  3
                </div>
              </div>
            </label>
          </styled.div>
        </li>
        <li
          key="26"
        >
          <styled.div>
            <input
              checked={false}
              id="2020-02-04"
              name="date"
              onChange={[Function]}
              type="radio"
              value="2020-02-04"
            />
            <label
              htmlFor="2020-02-04"
            >
              <div>
                <div>
                  火
                </div>
                <div>
                  4
                </div>
              </div>
            </label>
          </styled.div>
        </li>
        <li
          key="27"
        >
          <styled.div>
            <input
              checked={false}
              id="2020-02-05"
              name="date"
              onChange={[Function]}
              type="radio"
              value="2020-02-05"
            />
            <label
              htmlFor="2020-02-05"
            >
              <div>
                <div>
                  水
                </div>
                <div>
                  5
                </div>
              </div>
            </label>
          </styled.div>
        </li>
        <li
          key="28"
        >
          <styled.div>
            <input
              checked={false}
              id="2020-02-06"
              name="date"
              onChange={[Function]}
              type="radio"
              value="2020-02-06"
            />
            <label
              htmlFor="2020-02-06"
            >
              <div>
                <div>
                  木
                </div>
                <div>
                  6
                </div>
              </div>
            </label>
          </styled.div>
        </li>
        <li
          key="29"
        >
          <styled.div>
            <input
              checked={false}
              id="2020-02-07"
              name="date"
              onChange={[Function]}
              type="radio"
              value="2020-02-07"
            />
            <label
              htmlFor="2020-02-07"
            >
              <div>
                <div>
                  金
                </div>
                <div>
                  7
                </div>
              </div>
            </label>
          </styled.div>
        </li>
        <li
          key="30"
        >
          <styled.div>
            <input
              checked={true}
              id="2020-02-08"
              name="date"
              onChange={[Function]}
              type="radio"
              value="2020-02-08"
            />
            <label
              htmlFor="2020-02-08"
            >
              <div>
                <div>
                  土
                </div>
                <div>
                  8
                </div>
              </div>
            </label>
          </styled.div>
        </li>
        <li
          key="31"
        >
          <styled.div>
            <input
              checked={false}
              id="2020-02-09"
              name="date"
              onChange={[Function]}
              type="radio"
              value="2020-02-09"
            />
            <label
              htmlFor="2020-02-09"
            >
              <div>
                <div>
                  日
                </div>
                <div>
                  9
                </div>
              </div>
            </label>
          </styled.div>
        </li>
      </styled.ul>
    </div>
  </StyledCard>
  <Connect(HabitList)
    habitRecords={
      Immutable.Record {
        "items": Immutable.List [
          Immutable.Record {
            "habitId": 2,
            "completedAt": "2020-02-08T00:00:00.000Z",
            "isSkipped": false,
          },
          Immutable.Record {
            "habitId": 1,
            "completedAt": "2020-01-07T15:00:00.000Z",
            "isSkipped": false,
          },
        ],
      }
    }
    habits={
      Immutable.Record {
        "items": Immutable.List [
          Immutable.Record {
            "id": 1,
            "habitName": "読書する",
            "repeatType": "day_of_week",
            "repeatValue": 127,
            "startedAt": "2020-02-08T00:00:00.000Z",
            "targetTime": 0,
            "timeOfDay": "",
            "consecutiveDays": 0,
            "consecutiveWeeks": null,
          },
          Immutable.Record {
            "id": 2,
            "habitName": "筋トレ",
            "repeatType": "day_of_week",
            "repeatValue": 127,
            "startedAt": "2020-02-08T00:00:00.000Z",
            "targetTime": 0,
            "timeOfDay": "",
            "consecutiveDays": 0,
            "consecutiveWeeks": null,
          },
          Immutable.Record {
            "id": 3,
            "habitName": "新しいCDを1枚聴く",
            "repeatType": "day_of_week",
            "repeatValue": 127,
            "startedAt": "2020-02-08T00:00:00.000Z",
            "targetTime": 0,
            "timeOfDay": "",
            "consecutiveDays": 0,
            "consecutiveWeeks": null,
          },
        ],
      }
    }
    selectedDate="2020-02-08"
  />
</div>
`;
